### ISSUE TITLE
Allow nested values for DataTable header key

### ISSUE THREAD

**Issue by Trystan-SA** [IBM/carbon-components-svelte#564]
If I have data ordered like this : 

```js 
[{ 
    id: "123456"
    identity: {
       firstName: "John",
       lastName: "Doe"
    }
 }]
```

It's impossible to select "lastName" as a key

```js

<DataTable
   headers={[
      {value:"Last Name"  key: "identity.lastName"} ???
   ]}
/>
```

The values inside the DataTable will be undefined.

**Comment by Trystan-SA** [comment]
If I use the **display** and **sort** parameters, I can make it run almost correctly.

```js 
<DataTable
   headers={[
      {value:"Last Name", key: "identity", 
      display: (identity)=> identity.lastName, 
      sort:(a,b)=>{ 
         if(a.lastName > b.lastName){return 1}
         else if(a.lastName < b.lastName{ return -1}
      }
   }]
/>
```

But if I want to have the same thing with the firstName field, a glitch will appear. This will mix the order of the column when sorting them (probably because they have the same key as reference) 

- When I sort any column for the first time (the result is the same for all of them) . the lastName column will get the values of the firstName column.
- When I sort it a second time. The batchExpansion column will get the value of the firstName column. The firstName column will get the value of the batchExpansion column (an arrow on every field) and the lastName column stay with the values of the firstName column.
- Nothing changes if I do it again.


So I don't think there is another option for me (apart from changing the value of the array directly)

**Comment by Trystan-SA** [comment]
I found a way to fix my issue.

I can flatten my JS object (by using the [JS flat package](https://www.npmjs.com/package/flat) or doing it myself)
It will then work as expected.

The issue itself is not solved, but this is a nice workaround.

**Comment by alkanna** [comment]
Hey @Trystan-SA, a PR was just merged that fixes this issue : #602

**Comment by Trystan-SA** [comment]
Great work ! Thank you for this PR :)
